testsuite/tools/expect_test.ml

```diff
@@ -315,6 +315,17 @@ let process_expect_file fname =
 let repo_root = ref None
 let keep_original_error_size = ref false
 
+let always_read_topdir_signature root =
+  let compiler_libs =
+    Filename.concat Config.standard_library "compiler-libs" in
+  let installed_topdirs_cmi = Filename.concat compiler_libs "topdirs.cmi" in
+  if Sys.file_exists installed_topdirs_cmi then
+    () (* we have read topdirs from the installation directory *)
+  else
+    let in_tree_topdirs_cmi =
+      Filename.concat root @@ Filename.concat "toplevel"  "topdirs.cmi" in
+    ignore (Env.read_signature "Topdirs" in_tree_topdirs_cmi)
+
```

Comment:
I'm not sure I understand how this works.
I would have expected priority to be given to the in tree file, but apparently priority is given to the installed one.
What happens if the installed file is obsolete?

Reply:
What happens is that none of the topdirs.cmi files take priority currently: both files are always loaded, and inconsistent interfaces result in a `Consistbl.Make(Module_name).Inconsistency("Topdirs", ...)` error. This hack make ensure that we are always loading twice `topdirs.cmi`.

Reply:
The same issue affects the `runtop` target for launching a development toplevel. Definitely a bug to fix during the 5.0-cycle, but only affecting the development tree.